localstack-core/localstack/services/stepfunctions/provider.py

```diff
@@ -1057,19 +1057,40 @@ def list_state_machine_aliases(
         max_results: PageSize = None,
         **kwargs,
     ) -> ListStateMachineAliasesOutput:
-        # TODO: add pagination support.
+        assert_pagination_parameters_valid(max_results, next_token)
+
         self._validate_state_machine_arn(state_machine_arn)
         state_machines = self.get_store(context).state_machines
         state_machine_revision = state_machines.get(state_machine_arn)
         if not isinstance(state_machine_revision, StateMachineRevision):
             raise InvalidArn(f"Invalid arn: {state_machine_arn}")
 
         state_machine_aliases: StateMachineAliasList = list()
+        valid_token_found = next_token is None
+
         for alias in state_machine_revision.aliases:
             state_machine_aliases.append(alias.to_item())
+            if alias.get_token() == next_token:
+                valid_token_found = True
+
+        if not valid_token_found:
+            raise InvalidToken("Invalid Token: 'Invalid token'")
```

Comment:
Kudos for checking the token validity! I guess some logic can go into `PaginatedList` to make it available for all services. It would probably be a stretch for this PR as I imagine not all the services would raise an exception though.